### DISCUSSION TITLE
How to use pipes with $os.cmd()?

### DISCUSSION THREAD

**Discussion by AfdhalAtiffTan** [pocketbase/pocketbase#4946] [Q&A] [answered]
Hello,

For example, this command:
```bash
ls | grep old
```

I am not sure how to get the output into a variable:
```js
const cmd_a = $os.cmd('ls');
const cmd_b = $os.cmd('grep', 'old');

cmd_b.stdin = cmd_a.stdoutPipe();
// cmd_b.stdout = os.stdout; // ?

cmd_b.start();
cmd_a.run();
cmd_b.wait();
```

Any suggestions?

**Comment by ganigeorgiev** [comment, marked as answer]
I'm not sure how this could be easily done with the JSVM since we currently don't have a helper that will convert a writer into a string/bytes slice.

A simpler alternative could be to invoke the `sh -c` command instead, ex. `$os.cmd('sh', '-c', 'ls | grep old')`

**Reply by AfdhalAtiffTan** [reply]
That has worked for me, thank you!
